### PR TITLE
Payment Activity Card – display disputes and refunds as positive numbers

### LINKED ISSUE #8748: Payment Activity card Refunds & Disputes amounts should show as positive values
Part of epics #8487 #8488

The Payment Activity card Refunds/Disputes/Fees amounts should all show as positive values, as per the design. 

Disputes and Refunds currently show as negative values.

Image

See discussion pdjTHR-3DQ-p2#comment-4966

### PR DESCRIPTION
**Pull request by Jinksi**
Fixes #8748

> [!NOTE]  
> The solution for the linked issue may be best solved on server (see [comment below](https://github.com/Automattic/woocommerce-payments/pull/8749#pullrequestreview-2035613192)). This PR will be adapted as necessary (potentially only introducing/updating integration tests).

#### Changes proposed in this Pull Request

This PR ensures the Refunds/Disputes amounts render as a positive value, as per the design.

Disputes and Refunds are stored as negative values and are returned from the server API as such.

image

#### Testing instructions

- Enable the feature flag with `wp option update _wcpay_feature_payment_overview_widget 1` in CLI (`npm run cli` if docker)
- Purchase 3+ orders, one with a dispute test card (`4000000000000259`).
- Create a refund for one order via the order edit screen.
- View `Payments → Overview` and observe the Payment Activity Card.
- You should see values in all tiles, similar to the screenshot above.
- Ensure all values are displayed as positive numbers.

-------------------

- [x] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times. 
- [x] Covered with tests (or have a good reason not to test in description ☝️)
- [x] Tested on mobile **N/A**

**Post merge**

- [ ] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : _Add link here / 'QA Testing Not Applicable'_
- [ ] Add or update [critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) and [testing instructions for critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows), if applicable.
- [ ] Add what's changed (description, screenshot, demo videos etc.) to the release announcement post, if applicable.

### CODE REVIEW

File: client/components/payment-activity/payment-activity-data.tsx [hunk 1]
```diff
@@ -36,12 +36,13 @@ const PaymentActivityData: React.FC = () => {
 		getDateRange()
 	);
 
+	const currency = paymentActivityData?.currency;
```
Comment (Jinksi) [inline client/components/payment-activity/payment-activity-data.tsx:39]: This is just re-organising to keep related vars adjacent to one another.

**Comment by botwoo** [comment]
### Test the build 

#### Option 1. Jetpack Beta

- Install and activate [Jetpack Beta](https://jetpack.com/download-jetpack-beta/).
- Use this build by searching for PR number `8749` or branch name `fix/8748-payment-activity-card-positive-amount-values` in your-test.site/wp-admin/admin.php?page=jetpack-beta&plugin=woocommerce-payments          

#### Option 2. Jurassic Ninja - available for logged-in A12s

:rocket: [Launch a JN site with this branch](https://jurassic.ninja/create/?jetpack-beta&shortlived&nojetpack&woocommerce&woocommerce-payments-dev-tools&branches.woocommerce-payments=fix/8748-payment-activity-card-positive-amount-values) :rocket:

:information_source: Install this [Tampermonkey script](https://github.com/Automattic/woocommerce-payments/tree/develop/bin/wcpay-live-branches) to get more options.

---

Build info: 

- Latest commit: 57c11d627dd8036a54aa2e5dd1ea7b1bcc787414
- Build time: 2024-05-07 03:21:00 UTC

_Note: the build is updated when a new commit is pushed to this PR._

File: client/components/payment-activity/test/index.test.tsx [hunk 2]
```diff
@@ -129,9 +129,17 @@ describe( 'PaymentActivity component', () => {
 		// Check survey is rendered.
 		getByText( 'Are these metrics helpful?' );
 
-		// Check correct currency/value is displayed.
+		// Check TPV value is displayed with the correct currency symbol and format.
 		const tpvElement = getByLabelText( 'Total payment volume' );
-		expect( tpvElement ).toHaveTextContent( '€1.234,56' );
+		expect( tpvElement ).toHaveTextContent( /^€1.234,56/ );
```
Comment (Jinksi) [inline client/components/payment-activity/test/index.test.tsx:135]: I opted for regex with the start-of-string `^` matcher to ensure the value isn't prefixed with `-`.

Comment (haszari) [reply client/components/payment-activity/test/index.test.tsx:135]: This explanation of intent would make a great code comment, for future test-updaters :)

Comment (Jinksi) [reply client/components/payment-activity/test/index.test.tsx:135]: Good idea → added in 2c4bd52dd54daefae5c2d9555ae20df4095a4f62

**Comment by jessy-p** [comment]
Should the API return positive numbers? I don't think there is any issue with doing that.

**Review by jessy-p — COMMENTED** [review]
Should the API return positive numbers? I don't think there we have any issue with doing that, so we can avoid modifying on front-end.

**Comment by Jinksi** [comment]
> Should the API return positive numbers

It might be best to look at other API endpoints and how refund/dispute/fee amounts are returned and keep the pos/neg consistent. We may have this preference documented somewhere.

Part of me prefers the backend to show the "true" pos/neg form of the value and let the UI do the manipulation, similar to currency formatting.

Also, we will want these values as negative values for the chart, so the UI will need to manipulate either way.

The other part of me wants all the values to be positive from the API endpoint, for consistency within the endpoint response (e.g. why isn't fees a negative?). 🤔

**Comment by jessy-p** [comment]
> It might be best to look at other API endpoints and how refund/dispute/fee amounts are returned and keep the pos/neg consistent.

The other APIs are individual APIs for disputes etc and have the numbers as positive.

**Comment by jessy-p** [comment]
I am currently preferring API returning positive values for all values, rather than returning negative values for disputes and manipulating in FE to make it positive for the following reasons
- consistency with other APIs which format these amounts (disputes, refunds) as positive
- defaulting all values to positive and further presentation changes by FE based on need is straightforward. The other way (some positive and some negative) requires devs to keep that in mind for each endpoint and switch presentation accordingly.

@Automattic/transact-architecture

**Comment by Jinksi** [comment]
I agree, positive values make more sense when talking about the context of this API response – as long as we document this clearly in the API docs and/or code / function docblock.

Looking to Stripe API for some inspiration, they show positive integers when referring to the Dispute/Refund object amount and the Balance Transaction can be observed to calculate the net difference (pos or neg).

* [A `Refund` API response will return a positive integer](https://docs.stripe.com/api/refunds/object#refund_object-amount)
* The related `Balance Transaction` will show the [negative gross/net amount](https://docs.stripe.com/api/balance_transactions/object#balance_transaction_object-amount)

**Comment by Jinksi** [comment]
☝️ I'll prepare a server issue and PR, adapting this PR as necessary (may keep the integration tests).

**Comment by jessy-p** [comment]
> The related Balance Transaction will show the [negative gross/net amount](https://docs.stripe.com/api/balance_transactions/object#balance_transaction_object-amount)

This explains why we get (and store) negative values, we use that API for Transactions cache.

**Comment by jessy-p** [comment]
Closing in favor of server-side solution https://github.com/Automattic/woocommerce-payments-server/pull/5287

**Pull request closed without merging**